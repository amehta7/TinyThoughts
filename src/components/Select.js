import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const Select = ({ categories, label, value, onSet, hasError }) => {
  return (
    <div className={hasError ? 'select-list hasError' : 'select-list'}>
      <span className='label'>{label}</span>
      <select value={value} onChange={onSet}>
        {categories.map(({ id, title }) => {
          return (
            <option key={id} id={id}>
              {title}
            </option>
          )
        })}
      </select>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    categories: state.posts.categories,
  }
}

Select.propTypes = {
  type: PropTypes.string,
}

export default connect(mapStateToProps, null)(Select)
