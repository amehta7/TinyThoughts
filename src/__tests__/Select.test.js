import React from "react";
import { shallow } from "enzyme";
import Select from "../components/Select";

describe("Select component", () => {
  it("Renders a controlled component", () => {
    const onSetFn = jest.fn();
    const wrapper = shallow(
      <Select
        label="Pick a fruit"
        options={[
          { id: 1, title: "Apple" },
          { id: 2, title: "Kiwi" }
        ]}
        onSet={onSetFn}
        hasError={false}
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find("select").simulate("change", { target: { value: 2 } });
    expect(onSetFn).toHaveBeenCalledTimes(1);
  });

  it("Renders a red border when hasError is set to true", () => {
    const wrapper = shallow(
      <Select
        label="Pick a fruit"
        options={[
          { id: 1, title: "Apple" },
          { id: 2, title: "Kiwi" }
        ]}
        onSet={() => {}}
        hasError={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
