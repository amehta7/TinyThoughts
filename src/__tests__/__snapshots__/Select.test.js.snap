// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a controlled component`] = `
<div
  className="select-list"
>
  <span
    className="label"
  >
    Pick a fruit
  </span>
  <select
    defaultValue="Select"
    onChange={[MockFunction]}
  >
    <option
      disabled={true}
    >
      Select
    </option>
    <option
      key="1"
      value={1}
    >
      Apple
    </option>
    <option
      key="2"
      value={2}
    >
      Kiwi
    </option>
  </select>
</div>
`;

exports[`Renders a red border when hasError is set to true`] = `
<div
  className="select-list hasError"
>
  <span
    className="label"
  >
    Pick a fruit
  </span>
  <select
    defaultValue="Select"
    onChange={[Function]}
  >
    <option
      disabled={true}
    >
      Select
    </option>
    <option
      key="1"
      value={1}
    >
      Apple
    </option>
    <option
      key="2"
      value={2}
    >
      Kiwi
    </option>
  </select>
</div>
`;
